sync/src/synchronizer/headers_process.rs

```diff
@@ -145,15 +145,18 @@ impl<'a> HeadersProcess<'a> {
         }
 
         if headers.is_empty() {
-            // Reset headers sync timeout
-            self.synchronizer
-                .peers()
-                .state
-                .write()
-                .get_mut(&self.peer)
-                .expect("Peer must exists")
-                .headers_sync_timeout = None;
-            debug!("HeadersProcess is_empty (synchronized)");
+            debug!("HeadersProcess is_empty");
+            let ibd = self.active_chain.is_initial_block_download();
+            if !ibd {
```

Comment:
Since I had implement a mechanism base on the sync speed, I thought we can allow keep the connection here.
We don't reject any nodes for one single fault, we measure their credit base on a period of time.

I had confirmed this part with @TheWaWaR last week. (Maybe I made a mistake on understanding his explanation,) I thought may be this message just means that I had reach the tip.